[REF] Subscriptions: rewrite overview, integrate products page (#12823)
odoo/documentation: content/applications/sales/subscriptions.rst

:show-content:
:hide-toc:

=============
Subscriptions
=============

The Odoo **Subscriptions** app is designed to manage recurring revenue through subscription-based
products or services. It supports automated invoicing, renewal management, and customer lifecycle
tracking.

Subscriptions can be created manually or automatically through online sales, with varying options
for recurring billing. The app integrates with other Odoo modules such as Invoicing, CRM, Sales, and

> Helpdesk to support end-to-end subscription workflows.

Author: I see what you mean. AFAIK we use these terms interchangeably so it's a matter of preference. Tomato tomato ^^

.. cards::

   .. card:: Subscription plans
      :target: subscriptions/plans
      :large:

      Customize subscription plan templates tailored to various product offerings

   .. card:: Renew a subscription
      :target: subscriptions/plans
      :large:

      Understand the core management activity for subscriptions

   .. card:: Upsell a subscription
      :target: subscriptions/plans
      :large:

      Offer more value for current subscribers on the same sales order

   .. card:: Integrate subscriptions with eCommerce
      :target: subscriptions/plans
      :large:

> 

Reviewer: These cards are awesome!

      Offer subscription products with an Odoo eCommerce app integration

.. seealso::

>    - `Odoo Tutorials: Subscriptions <https://www.odoo.com/slides/subscription-20>`_

Reviewer: Never occurred to me to link to the TUTORIALS!!! mind. blown.

Recurrence periods are the time windows in which subscriptions are active before they renew again.
While a subscription is active, customers receive products or services, and may also have access to
additional benefits such as support desk triage. In terms of payment, these recurrence periods
designate how often the customer is charged in order to maintain the benefits of their subscription.

To configure recurrence periods, go to :menuselection:`Subscriptions app --> Configuration -->
Recurrence periods`.

By default, the **Subscriptions** app includes a number of common recurrence periods already
available, such as :guilabel:`Monthly` and :guilabel:`Yearly`.

Create new recurrence periods by clicking :guilabel:`NEW` on the :guilabel:`Recurrence Periods`
dashboard, to reveal a blank form where the period :guilabel:`Name`, :guilabel:`Duration` and
:guilabel:`Unit` values are specified.

.. image:: subscriptions/recurrence-period-blank-form.png
   :align: center
   :alt: A blank recurrence period form in the Odoo Subscriptions application.

.. important::
   The unit :guilabel:`Days` *cannot* be used as a recurrence period for subscription products. The

>    daily recurrence period in Odoo is designated for rentals, and **cannot** be added to

Reviewer: ```suggestion
   daily recurrence period in Odoo is designated for rentals **only**, and **cannot** be added to
```
just for more emphasis

Author: If we get folks submitting tickets suggesting that they missed or misunderstood this, then we can add more emphasis...I do think the `important` block + the bold *cannot* should get the point across (or at least I hope 😛 )

   subscription-based sales orders.

   This limitation is there to avoid sales orders that would generate daily invoices.

Product form configuration
==========================

With recurrence periods set up, create a subscription product by navigating to
:menuselection:`Subscriptions app --> Subscriptions --> Products`, and either clicking an existing
product to turn into a subscription, or by clicking :guilabel:`NEW` to open up a blank product form.

.. note::
   By default, the :guilabel:`Recurring` option is already enabled, prompting Odoo to recognize it
   as a subscription product. Be sure to leave the :guilabel:`Recurring` and :guilabel:`Can be Sold`
   options enabled.

.. image:: subscriptions/subscription-product-form.png
   :align: center
   :alt: A basic subscription product form in Odoo Subscriptions application.

> On the product form, configure the following items on the :guilabel:`General Information` tab so the

Reviewer: ```suggestion
On the product form, configure the following items in the :guilabel:`General Information` tab so the
```

subscription product will function correctly:

- :guilabel:`Recurring`: check this box to turn the product into a subscription in Odoo.
- :guilabel:`Product type`: this value is typically set to a :guilabel:`Service`, however other
  product types may be used (e.g., physical product box subscriptions, eLearning course, etc.).

> - :guilabel:`Invoicing policy`: set this value to when the customer should be charged for their

Reviewer: The options for the various 'invoicing policies' are so different, I suggest either adding a sub-section where the options are explained, with perhaps examples of when/why you would choose each one, or if they are explained in a different section, to link them, so readers can jump to it.

Author: Good call, found [a doc](https://www.odoo.com/documentation/16.0/applications/sales/sales/invoicing/invoicing_policy.html) that explains a few options.

  subscription.
- :guilabel:`Unit of Measure`: how the product should be counted in Odoo, for stock purposes. For

>   most subscriptions, the :abbr:`UoM (Unit of Measure)` will be :guilabel:`:Units`.

Reviewer: ```suggestion
  most subscriptions, the :abbr:`UoM (Unit of Measure)` will be :guilabel:`Units`.
```

Author: Whoops, good catch! ^^

- :guilabel:`Sales Price`: enter the recurring cost of the subscription that the customer will pay
  per recurrence period.

> Optionally set up information on the :guilabel:`Attributes & Variants` tab if the subscription

Reviewer: I don't know if there's an appropriate doc to link to the 'Attributed & Variants' words. I looked, and they seem more general - but if you think they would help/apply to this, I would link them. I feel like there must be some place in the docs that address this particular istuation.

Author: Yup, there's [another doc](https://www.odoo.com/documentation/16.0/applications/sales/sales/products_prices/products/variants.html) for this one too. Will link it as well!

contains multiple choices for customers (i.e. food delivery, tailored fashion boxes, etc.).

In the :guilabel:`Time-based pricing` tab, clarify the pricing options for the subscription. For
each option available, click :guilabel:`Add a price` to add a new row.

.. tip::
   Longer time :guilabel:`Period` options are typically incentivized with cost savings. Consider

>    dropping the total :guilabel:`Price` value(s) to offer customers a discount while supporting the

Reviewer: ```suggestion
   dropping the total :guilabel:`Price` values to offer customers a discount while supporting the
```

I believe we are supposed to use just the plural when we need to use this format? (Although it's been months since I wrote a doc as I have been in tutorial heaven!)

Author: Seems okay since subscriptions periods are likely to have more than one option.

   business's financial runway.

> Last, if this subscription is meant to be purchased on the **eCommerce** website, click the

Reviewer: ```suggestion
Last, if the subscription is meant to be purchased on the **eCommerce** website, click the
```

:icon:`fa-globe` :menuselection:`Go To Website` smart button and in the product page header, click
the gray slider from :guilabel:`Unpublished` to the green :guilabel:`Published` status.

.. _subscriptions/quotations:

Create a subscriptions quotation
================================

Manually create a new customer subscription by navigating to either the :menuselection:`Sales` or
:menuselection:`Subscriptions` app dashboards, and then clicking :guilabel:`NEW`.

.. note::
   Products that have been marked as :guilabel:`Recurring` on their product forms, and are also sold
   on the **eCommerce** website will *automatically* create and confirm subscription quotations in
   the backend of Odoo.

.. important::
   Sales orders with a defined recurrence period automatically become subscriptions.

On the quotation form, fill in the necessary field information such as :guilabel:`Customer` and the
:guilabel:`Order Lines` tab.

Optionally, specify a:

> - :doc:`Quotation Template </applications/sales/subscriptions/plans>`, if one is readily available

Reviewer: I found this confusing since you go to this quotation template doc, then you go to subscriptions pans, but then click again on 'quotation templates'

Form my read, it sounds like a quotation template IS a configured subscription product? I feel like a sentence or two explaining that (if that is the case) could help. Then you can link the 'subscription plan', and it would make more immediate sense. Just a thought, since I was a bit confused.

Author: Yea I understand the confusion.
After Odoo 17, these were known as "Quotation Templates" but before Odoo 17, they were called "Subscription Plans." I'll try to clarify the language a bit...

Author: Better link we can use, for now: https://www.odoo.com/documentation/16.0/applications/sales/sales/send_quotations/quote_template.html

  to help populate the form fields.
- :guilabel:`Expiration` date, to indicate when the subscription offer is no longer valid.

  .. tip::
     Expiration dates pair well with :ref:`discounts <sales/products_prices/discounts>` to
     incentivize faster purchases, since the discount will expire with the quotation if it's not
     turned into a sales order within the specified date range.

- :ref:`Pricelist <sales/product_prices/pricelist>`, if one is available and appropriate to use

>   (i.e., summer sale discount, VIP customer, etc.).

Reviewer: The only field in this little 'chunk' that was not called ut was 'Recurrence'. Is that because it's not 'optional'? I feel like it should be called out since it's the only one left out.

Author: Good catch. Recurrence is a mandatory field, I added it back on line 137
`On the quotation form, fill in the necessary fields such as :guilabel:`Customer` and
:guilabel:`Recurrence`, as well as the :guilabel:`Order Lines` tab.`

>    Click on :guilabel:`Customer Preview` to preview the customer portal where the customer can view

Reviewer: General note for the lines above (it won't let me click on the line for some reason). The buttons are in all caps, 'SEND BY EMAIL' and 'CONFIRM.' I see that when you click 'NEW' it is written in all caps, as it appears on the UI. So to be consistent, I would update these to be in all caps (either that, or revert the 'NEW' buttons to 'New' buttons).

Author: fixed!
